### PR TITLE
fixing bug in wcssstr

### PR DESCRIPTION
**Pull request by mike-kaufman**
Fixing https://github.com/Microsoft/ChakraCore/issues/3434..  Bug in wcsstr would return a match if the string being searched ended with a prefix of the search string.  Fixed this to behave correctly.  Also added tests to test_native.sh to verify this.

### CODE REVIEW

File: test/native-tests/test-pal/sample.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "pal.h"
+
+int wscmp(const char16_t *left, const char16_t *right)
```
Comment (obastemur) [inline test/native-tests/test-pal/sample.cpp]: why don't we use PAL_wcscmp?

Comment (mike-kaufman) [inline test/native-tests/test-pal/sample.cpp]: you're right, for some reason I thought this was removed from our copy.  I'll fix...

File: test/native-tests/test_native.sh [hunk 2]
```diff
@@ -42,13 +47,7 @@ TEST () {
...
 }
 
-RES=$(c++ --version)
-if [[ ! $RES =~ "Apple LLVM" ]]; then
-    FIND_CLANG
-else
-    CC="cc"
-    CXX="c++"
-fi
+FIND_CLANG
```
Comment (obastemur) [inline test/native-tests/test_native.sh]: why this change?

Comment (mike-kaufman) [inline test/native-tests/test_native.sh]: including pal.h causes a bunch of errors around `__declspec` syntax, so you need `-fms_extensions` flag passed to clang.

Comment (obastemur) [inline test/native-tests/test_native.sh]: well, if we remove this, we won't be able to test gcc/g++ compiling a test app. You don't need to include PAL.h to use that. define the extern inside the C/xx file and linker will find it for you.

File: test/native-tests/test-pal/sample.cpp (hunk 1 above)
Comment (mike-kaufman) [inline test/native-tests/test-pal/sample.cpp]: fixed now.

File: test/native-tests/test_native.sh (hunk 2 above)
Comment (mike-kaufman) [inline test/native-tests/test_native.sh]: fixed now.

File: test/native-tests/test_native.sh [hunk 3]
```diff
@@ -1,3 +1,5 @@
+#!/bin/bash
+
```
Comment (obastemur) [inline test/native-tests/test_native.sh]: this breaks the Copyright check (CI)

**Comment by mike-kaufman** [comment]
Closing, will re-open though chakra-hub.
